testutils/infra/container_manager/docker_compose_base_manager.py

```diff
@@ -181,10 +181,34 @@ def _docker_compose_cmd(self, arg_list, env=None):
 
     def _stop_docker_compose(self):
         with docker_lock:
+            output = subprocess.check_output(
+                "docker ps -f name=%s" % self.name, shell=True
+            ).decode()
+            logger.info("%s: stopping all:\n%s\n" % (self.name, output))
+
             stop_sleep_seconds = 15
             retry_attempts = 8
 
             # Take down all docker instances in this namespace.
+            # QA-455 special treatment for redis
+            while retry_attempts > 0:
+                logger.info("qa-455: %s: attempting to shutdown redis" % self.name)
+                cmd = (
+                    "docker ps -f name=%s | grep -F redis | awk '{print($1);}' | xargs -n1 -I{} docker exec {} redis-cli shutdown"
```

Comment:
Ole, when you see that, the first thing you will say: "you could have done it all in awk". I bet :)